26_CentralLimitBoxBlur/main.cpp

```diff
@@ -0,0 +1,432 @@
+			smart_refctd_ptr<nbl::video::IGPUImage>& imgOut,
+			smart_refctd_ptr<nbl::video::IGPUImageView>& imgViewOut
+		) {
+			video::IGPUImage::SCreationParams gpuImageCreateInfo;
+			gpuImageCreateInfo.flags = inCpuTexInfo.flags;
+			gpuImageCreateInfo.type = inCpuTexInfo.type;
+			gpuImageCreateInfo.extent = inCpuTexInfo.extent;
+			gpuImageCreateInfo.mipLevels = inCpuTexInfo.mipLevels;
+			gpuImageCreateInfo.arrayLayers = inCpuTexInfo.arrayLayers;
+			gpuImageCreateInfo.samples = inCpuTexInfo.samples;
+			gpuImageCreateInfo.tiling = video::IGPUImage::TILING::OPTIMAL;
+			gpuImageCreateInfo.usage = usageFlags | asset::IImage::EUF_TRANSFER_DST_BIT;
+			gpuImageCreateInfo.queueFamilyIndexCount = 0u;
+			gpuImageCreateInfo.queueFamilyIndices = nullptr;
+
+			gpuImageCreateInfo.format = m_physicalDevice->promoteImageFormat(
+				{ inCpuTexInfo.format, gpuImageCreateInfo.usage }, gpuImageCreateInfo.tiling
+			);
+			auto gpuImage = m_device->createImage( std::move( gpuImageCreateInfo ) );
+
+			auto gpuImageMemReqs = gpuImage->getMemoryReqs();
+			gpuImageMemReqs.memoryTypeBits &= m_physicalDevice->getDeviceLocalMemoryTypeBits();
+			m_device->allocate( gpuImageMemReqs, gpuImage.get(), video::IDeviceMemoryAllocation::EMAF_NONE );
+
+			auto imgView = m_device->createImageView( {
+				.flags = IGPUImageView::ECF_NONE,
+				.subUsages = usageFlags,
+				.image = gpuImage,
+				.viewType = IGPUImageView::ET_2D,
+				.format = gpuImageCreateInfo.format
+			} );
+			gpuImage->setObjectDebugName( name.data() );
+			imgView->setObjectDebugName( ( std::string{ name } + "_view" ).c_str() );
+			imgOut = gpuImage;
+			imgViewOut = imgView;
+		};
+
+
+		smart_refctd_ptr<nbl::video::IGPUImage> inputGpuImg;
+		smart_refctd_ptr<nbl::video::IGPUImage> outputGpuImg;
+		smart_refctd_ptr<nbl::video::IGPUImageView> inputGpuImgView;
+		smart_refctd_ptr<nbl::video::IGPUImageView> outputGpuImgView;
+		createGPUImages( IGPUImage::EUF_SAMPLED_BIT, "InputImg", inputGpuImg, inputGpuImgView );
+		createGPUImages( IGPUImage::EUF_STORAGE_BIT, "OutputImg", outputGpuImg, outputGpuImgView );
+		assert(inputGpuImg&&outputGpuImg&&inputGpuImgView&&outputGpuImgView);
+
+		auto computeMain = checkedLoad.operator()< nbl::asset::ICPUShader >( "app_resources/main.comp.hlsl" );
+		smart_refctd_ptr<ICPUShader> overridenUnspecialized = CHLSLCompiler::createOverridenCopy(
+			computeMain.get(), 
+			"#define WORKGROUP_SIZE %s\n#define PASSES_PER_AXIS %d\n#define AXIS_DIM %d\n",
+			std::to_string( WorkgroupSize ).c_str(), PassesPerAxis, AxisDimension
+		);
+		smart_refctd_ptr<IGPUShader> shader = m_device->createShader( overridenUnspecialized.get() );
+		if( !shader )
+		{
+			return logFail( "Creation of a GPU Shader to from CPU Shader source failed!" );
+		}
+
+
+		// TODO: move to shaderd cpp/hlsl descriptors file 
+		// No because this is a C++ only struct, only move the binding and count there
+		NBL_CONSTEXPR_STATIC nbl::video::IGPUDescriptorSetLayout::SBinding bindings[] = {
+			{
```

Comment:
also a pro style tip, push parameters to init calls inside an extra scope, this way its easier for Intellisense (and me) to see what varialbes could be in use

Also use `const` liberally.